src/traces/isosurface/attributes.js

```diff
@@ -0,0 +1,184 @@
+var baseAttrs = require('../../plots/attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = extendFlat({
+    x: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the X coordinates of the vertices on X axis.'
+        ].join(' ')
+    },
+    y: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the Y coordinates of the vertices on Y axis.'
+        ].join(' ')
+    },
+    z: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the Z coordinates of the vertices on Z axis.'
+        ].join(' ')
+    },
+    volume: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'Sets the 4th dimension of the vertices.'
+        ].join(' ')
+    },
+    isocap: {
+        valType: 'boolean',
+        role: 'info',
+        editType: 'calc',
+        dflt: false,
+        description: [
+            'Enable caps between minimum and maximum iso values.'
+        ].join(' ')
+    },
+    isovalue: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc',
+        dflt: [0],
+        description: [
+            'Sets iso surface boundaries.'
+        ].join(' ')
+    },
+    meshalgo: {
+        valType: 'string',
+        role: 'info',
+        dflt: 'MarchingCubes',
+        editType: 'calc',
+        description: [
+            'Sets the isosurface polygonizer algorithm:',
+            'including: `MarchingCubes` (i.e. default),',
+            '`MarchingTetrahedra` or `SurfaceNets`.'
+        ].join(' ')
+    },
+
+    text: {
+        valType: 'string',
+        role: 'info',
+        dflt: '',
+        arrayOk: true,
+        editType: 'calc',
+        description: [
+            'Sets the text elements associated with the vertices.',
+            'If trace `hoverinfo` contains a *text* flag and *hovertext* is not set,',
+            'these elements will be seen in the hover labels.'
+        ].join(' ')
+    },
+
+    // Color field
+    color: {
+        valType: 'color',
+        role: 'style',
+        editType: 'calc',
+        description: 'Sets the color of the whole isosurface'
+    },
+    vertexcolor: {
+        valType: 'data_array',
+        role: 'style',
+        editType: 'calc',
+        description: [
+            'Sets the color of each vertex',
+            'Overrides *color*.'
+        ].join(' ')
+    },
+    facecolor: {
+        valType: 'data_array',
+        role: 'style',
+        editType: 'calc',
+        description: [
+            'Sets the color of each face',
+            'Overrides *color* and *vertexcolor*.'
+        ].join(' ')
+    },
+    transforms: undefined
+},
+
+colorscaleAttrs('', {
+    colorAttr: '`intensity`',
```

Comment:
Should be `'value'` here.

Reply:
This `intensity` array is filled internally by the `convert.js` and provided to `gl-mesh3d` module. So I thought we need to keep it this way.

Reply:
Used `value` instead of `intensity` there: https://github.com/plotly/plotly.js/pull/3340/commits/d9c1fa7e88309d6e500d2d12c178ba8c77b367d5